src/main/java/cloud/fogbow/rcs/core/intercomponent/RemoteFacade.java

```diff
@@ -0,0 +1,25 @@
+package cloud.fogbow.rcs.core.intercomponent;
+
+import cloud.fogbow.rcs.core.models.ServiceType;
+
+public class RemoteFacade {
+
+    private static RemoteFacade instance;
+    
+    private RemoteFacade() {}
+
+    public static RemoteFacade getInstance() {
+        synchronized (RemoteFacade.class) {
+            if (instance == null) {
+                instance = new RemoteFacade();
+            }
+            return instance;
+        }
+    }
+    
+    public String requestService(String senderId, ServiceType serviceType) {
+        // TODO call the requestService in the CatalogService
```

Comment:
why is this not making the call already?

Reply:
The idea was to avoid increasing the number of classes for the revision by not adding changes to the **CatalogService** class in this branch. But if you think it's better, I can bring the implementation to this branch.

Reply:
this change is big as-is so i'll wait for the next pr ;-)